nixbase32/nixbase32_test.go

```diff
@@ -0,0 +1,226 @@
+package nixbase32
```

Comment:
This was a separate test package to do the "black box testing" as described in https://pkg.go.dev/testing.

I also wanted to prevent certain test-related helpers to appear in the package itself.

Is there a reason this is discouraged now?

Reply:
It isn't discouraged. From a practical standpoint, I wanted to use the now-unexported `alphabet` constant in the tests, which is why I made the change.

Reply:
So the only reason on why we now have to move the tests into the package is because alphabet got unexported. As explained in https://github.com/nix-community/go-nix/pull/108#discussion_r1271294514, I'd prefer if it'd stay exported, and the tests a separate test package.

Reply:
Done.